__tests__/model.test.js

```diff
@@ -72,54 +74,86 @@ describe('Model', () => {
     });
   });
 
-  describe('#dispatch', () => {
-    test('provides a getter for store dispatch', () => {
-      expect(Model.dispatch).toBeInstanceOf(Function);
+  describe('Store', () => {
+    class Comment extends Model { static type = 'comments'; }
+    Registry.register(Comment);
+    class Person extends Model { static type = 'people'; }
+    Registry.register(Person);
+    Registry.store = mockStore;
+
+    describe('#dispatch', () => {
+      test('provides a getter for store dispatch', () => {
+        expect(Model.dispatch).toBeInstanceOf(Function);
+      });
     });
-  });
 
-  describe('#findById', () => {
-    test('should find an entity in store by id', () => {
-      const response = Model.findById(5, 'comments');
-      expect(response.body).toEqual(apiState.comments[5].body);
+    describe('#all', () => {
+      test('should return all values from a resource store', async () => {
+        const response = await Comment.all();
+
+        expect(response).toMatchSnapshot();
+      });
     });
-  });
 
-  describe('#findAll', () => {
-    test('should return all values from a resource store', () => {
-      const response = Model.findAll('comments');
+    describe('#findById', () => {
+      test('should find an entity in store by id', async () => {
+        const response = await Comment.findById(5);
+        expect(response.body).toEqual(apiState.comments[5].body);
+      });
+    });
+
+    describe('#where', () => {
+      test('should return an array of values matching a query', async () => {
+        const response = await Comment.where({ attributes: { body: 'First!' } });
```

Comment:
Good point! Though this would also close off the possibility to query attributes with joins. Then again, not sure that should be a frontend responsibility. What do you think?